code/modules/mob/living/carbon/xenomorph/castes/praetorian/abilities_praetorian.dm

```diff
@@ -467,3 +467,551 @@ GLOBAL_LIST_INIT(acid_spray_hit, typecacheof(list(/obj/structure/barricade, /obj
+	playsound(target, 'sound/weapons/punch1.ogg', 25, TRUE)
+
+	var/damage = xeno_owner.xeno_caste.melee_damage * xeno_owner.xeno_melee_damage_modifier
+	carbon_target.apply_damage(damage, BRUTE, target_limb ? target_limb : 0, MELEE)
+	carbon_target.apply_effect(0.1 SECONDS, WEAKEN) // So they go through mobs and what not.
+	if(ishuman(carbon_target) && carbon_target.stat == UNCONSCIOUS)
+		RegisterSignal(carbon_target, COMSIG_MOVABLE_MOVED, PROC_REF(on_movement_while_thrown))
+		RegisterSignal(carbon_target, COMSIG_MOVABLE_POST_THROW, PROC_REF(on_throw_end))
+	carbon_target.knockback(xeno_owner, 2, 2)
+
+	succeed_activate()
+	add_cooldown()
+
+/// Does brute/oxygen damage to unconscious humans if they move.
+/datum/action/ability/activable/xeno/dislocate/proc/on_movement_while_thrown(datum/source)
+	SIGNAL_HANDLER
+	var/mob/living/carbon/human/human_mob = source
+	if(human_mob.stat != UNCONSCIOUS)
+		UnregisterSignal(human_mob, COMSIG_MOVABLE_MOVED)
+		UnregisterSignal(human_mob, COMSIG_MOVABLE_POST_THROW)
+		return
+	if(human_mob.adjustOxyLoss(HUMAN_CRITDRAG_OXYLOSS))
+		return
+	INVOKE_ASYNC(human_mob, TYPE_PROC_REF(/mob/living/carbon/human, adjustBruteLoss), HUMAN_CRITDRAG_OXYLOSS)
+
+
+/// Removes signals for COMSIG_MOVABLE_MOVED and COMSIG_MOVABLE_POST_THROW for thrown humans.
+/datum/action/ability/activable/xeno/dislocate/proc/on_throw_end(datum/source)
+	SIGNAL_HANDLER
+	UnregisterSignal(source, COMSIG_MOVABLE_MOVED)
+	UnregisterSignal(source, COMSIG_MOVABLE_POST_THROW)
+
+// ***************************************
+// *********** Item Throw
+// ***************************************
+/datum/action/ability/activable/xeno/item_throw
+	name = "Item Throw"
+	action_icon_state = "item_throw"
+	action_icon = 'icons/Xeno/actions/praetorian.dmi'
+	desc = "Pick up a nearby item momentarily and throw it in a chosen direction. The item's size determines elements such as how fast or hard it hits."
+	ability_cost = 50
+	cooldown_duration = 12 SECONDS
+	keybinding_signals = list(
+		KEYBINDING_NORMAL = COMSIG_XENOABILITY_ITEM_THROW,
+	)
+	/// If they have moved at least a single tile since picking up an item. Used as a single-tile grace period before dropping.
+	var/single_tile_grace_period = TRUE
+	/// If we are holding an item.
+	var/obj/item/held_item
+	/// Mutable appearance of the held item.
+	var/mutable_appearance/held_appearance
+
+/datum/action/ability/activable/xeno/item_throw/can_use_ability(atom/A, silent = FALSE, override_flags)
+	. = ..()
+	if(!.)
+		return FALSE
+	// Any target is acceptable if we have an item.
+	if(held_item)
+		return TRUE
+	// If we do not, we only want items.
+	if(!isitem(A) || isgrabitem(A))
+		if(!silent)
+			A.balloon_alert(owner, "not an item")
+		return FALSE
+	var/obj/item/item_atom = A
+	if(item_atom.anchored)
+		if(!silent)
+			item_atom.balloon_alert(owner, "item is anchored")
+		return FALSE
+	if(!owner.Adjacent(item_atom))
+		if(!silent)
+			item_atom.balloon_alert(owner, "too far")
+		return FALSE
```

Comment:
be consistant in how you arrange these checks. Its a minor thing but you have the adjacency check first in the procs above (and thats more intuitive).